CDTDatastore/HTTP/CDTHTTPInterceptorContext.h

```diff
@@ -25,6 +25,11 @@ NS_ASSUME_NONNULL_BEGIN
 @property (readwrite, nonatomic, strong) NSMutableURLRequest *request;
 @property (nonatomic) BOOL shouldRetry;
 @property (nullable, readwrite, nonatomic, strong) NSHTTPURLResponse *response;
+/**
+ * For storing arbitrary per-context state
+ * NOTE: Users are strongly encouranged to use unique keys
+ */
+@property (readwrite, nonatomic, strong) NSMutableDictionary *state;
```

Comment:
We should also use generics here for those users who are using swift. I think `NSMutableDictionary<NSString*,NSObject*> *state` should do it.